[eBus] Fix expanded bytes

csowada opened the pull request.

watou commented:
The convert CR LF to LF commit might be the one causing conflicts.  Please rebase this PR to latest master, and please write up in detail what this problem is with existing ebus binding and what your fix does to correct it.  Thank you!

csowada commented:
Okay, rebased on the latest version.

watou commented on bundles/binding/org.openhab.binding.ebus/src/main/java/org/openhab/binding/ebus/internal/connection/EBusCommandProcessor.java:
```diff
@@ -160,7 +160,7 @@ public void setConfigurationProvider(EBusConfigurationProvider configurationProv
      * @return
      */
-    private byte[] composeEBusTelegram(String commandId, Byte dst, Byte src, Map<String, Object> values) {
+    public byte[] composeEBusTelegram(String commandId, Byte dst, Byte src, Map<String, Object> values) {
```
Not a critical point, but why increase the method visibility?

watou commented on bundles/binding/org.openhab.binding.ebus/src/main/java/org/openhab/binding/ebus/internal/utils/EBusUtils.java:
```diff
@@ -249,6 +249,11 @@ static public EBusTelegram processEBusData(byte[] data) {
 
+            if (uc_crc == (byte) 0xAA && crc == (byte) 0xA9 && data[++crcPos] == (byte) 0x01) {
+                // replace with original value
+                crc = (byte) 0XAA;
```
If you do another commit for this PR, please change to `0xAA` for consistency.

watou commented on bundles/binding/org.openhab.binding.ebus/src/main/java/org/openhab/binding/ebus/internal/utils/EBusUtils.java:
```diff
@@ -321,6 +326,11 @@ static public EBusTelegram processEBusData(byte[] data) {
 
+            if (uc_crc == (byte) 0xAA && crc == (byte) 0xA9 && data[++crcPos] == (byte) 0x01) {
+                // replace with original value
+                crc = (byte) 0XAA;
```
If you do another commit for this PR, please change to `0xAA` for consistency.

csowada commented:
So, I changed the two files and I have rebased the PR. I use this version since weeks in my productive environment. But I can ask in the community for feedback.

watou commented:
Thank you @csowada.  If there is any doubt in your mind that this might cause a regression for other users, please see if others can test it by posing at https://community.openhab.org.  Otherwise, this PR LGTM.

watou commented:
cherry picked onto 1.9.x
